### PR TITLE
[techdebt] Remove IsSiteOwned and related functionality

### PR DESCRIPTION
**Pull request by kopancek**
This is not needed anymore as all the external services are site owned now. Also removed related functionality which does not make sense anymore with site-level only external services

## Test plan

Existing unit tests, tested locally if SG still works and it seems it does.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.23% (-6.65 kb) 🔽 | -0.05% (-6.85 kb) | -0.00% (-0.20 kb) | 0.55% (+4) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/milan/remove-unused-code/compare-report.html;tab=live_object) for a full comparison between the commits b565907 and 41d1a47 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by mrnugget — APPROVED** [review]

**Comment by miveronese** [comment]
wow! this is a lot of code Had no idea that all of this wasn't necessary anymore.
Thanks for cleaning this up!

File: internal/repos/types.go [hunk 1]
```diff
@@ -125,86 +122,6 @@ func (r *RateLimitSyncer) SyncServices(ctx context.Context, services []*types.Ex
...
-			return nil, errors.Wrap(err, "getting scopes")
-		}
-		cache.Set(key, []byte(strings.Join(scopes, ",")))
-		return scopes, nil
-	default:
-		return nil, errors.Errorf("unsupported config type: %T", v)
-	}
-}
-
 func hashToken(token string) (string, error) {
```
Comment (eseliger) [inline internal/repos/types.go:125]: is `hashToken` still used?

**Review by eseliger — APPROVED** [review]

**Comment by ryanslade** [comment]
Part of https://github.com/sourcegraph/sourcegraph/issues/44700

**Pull request merged**